pkg/store/cache/caching_bucket_factory.go

```diff
@@ -30,6 +38,41 @@ type CachingBucketWithBackendConfig struct {
 	CachingBucketConfig CachingBucketConfig `yaml:"caching_config"`
 }
 
+type CachingBucketConfig struct {
+	// Basic unit used to cache chunks.
+	ChunkSubrangeSize int64 `yaml:"chunk_subrange_size"`
```

Comment:
Ok, so I am super confused why you did all this nice work of making the options typed for each operation and then here, we just accept EVERYTHING (: 

I would either stick to this (if you configure options for specific operation, skip unrelated fields) and then do the same in underlying code, OR let's do typed config here as well.


Since you started nice typed thing on code level, maybe it's worth to actually **reuse** those `getConfig` `iterConfig`  and parse that directly (add yaml tags there)? (: WDYT?

Reply:
> Ok, so I am super confused why you did all this nice work of making the options typed for each operation and then here, we just accept EVERYTHING (:

My thinking is this: I don’t think user should be configuring parameters of individual operations (get, iter, ...), which is what those small structs do. User-provided configuration is higher level, and it is then used to configure individual operations based on our expert knowledge of what Thanos is doing. (lines 113-122 in this file).

See for example how single MetafileExistsTTL is used in both Get and Exists operation config. I don’t think user should be able to do this.

Reply:
What do you mean by "we just accept EVERYTHING"? I explained my reasoning in previous comment, wdyt?

Reply:
I see. I missed the fact that this is hardcoded per funcionalities. Cool then.